src/statistics.jl

```diff
@@ -142,6 +142,58 @@ function Base.show(io::IO, method::GeodesicInterpolationWithinRadius)
     return print(io, "GeodesicInterpolationWithinRadius($(method.radius))")
 end
 
+"""
+    Statistics.cov(
+        M::AbstractManifold,
+        x::AbstractVector;
+        basis::AbstractBasis=DefaultOrthonormalBasis(),
+        tangent_space_covariance_estimator::CovarianceEstimator=SimpleCovariance(;
+            corrected=false,
+        ),
+        mean_estimation_method::AbstractEstimationMethod=GradientDescentEstimation(),
+        inverse_retraction_method::AbstractInverseRetractionMethod=default_inverse_retraction_method(
+            M,
+        ),
+    )
+
+Estimate the covariance matrix of points `x` on manifold `M`. Since the covariance matrix
+on a manifold is a rank 2 tensor, the function returns its coefficients in basis induced by
+the given tangent space basis. See Section 5 of [^Pennec2006] for details.
+
+The mean is calculated using the specified `mean_estimation_method`, and tangent vectors at
```

Comment:
Can we link here to the corresponding mean function? That way a user could see more about the estimation methods (otherwise link the `AbstractEstimationMethod` but I think linking the mean would be nicer)